src/main/java/frc/team2412/robot/subsystems/ShooterSubsystem.java

```diff
@@ -0,0 +1,61 @@
+package frc.team2412.robot.subsystems;
+
+import com.ctre.phoenix6.hardware.TalonFX;
+import com.revrobotics.CANSparkBase.ControlType;
+import com.revrobotics.CANSparkLowLevel.MotorType;
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.SparkAbsoluteEncoder;
+import com.revrobotics.SparkPIDController;
+import edu.wpi.first.math.util.Units;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.team2412.robot.Hardware;
+
+public class ShooterSubsystem extends SubsystemBase {
+	// CONSTANTS
+	// MOTOR SPEED VALUES
+	public static final double SPEAKER_SHOOT_SPEED = 0.8;
+	public static final double AMP_SHOOT_SPEED = 0.2;
+	public static final double ANGLE_CHANGE_SPEED = 0.15;
+
+	// HARDWARE
+	private final TalonFX shooterTopMotor;
```

Comment:
I would take away the final modifier from these hardware objects. Having final here will make it so that you are unable to modify the variables and thus even though it doesn't show any errors when you attempt to, you will not be able to actually set the speed of the motors, etc.

Reply:
Its good to use final on constants though, since they aren't supposed to be modified or changed

Reply:
Final only means that the variable can't be assigned to another value, so operations like setting the speed of the motors will still work. ([Here's an example](https://github.com/robototes/ChargedUp2023/blob/main/src/main/java/frc/team2412/robot/subsystems/ArmSubsystem.java#L163-L165) from last year where we used final on motor controller objects).